### PR TITLE
feat(icons): additions, deletions, modifications

### LINKED ISSUE #2035: icons: additions, deletions, modifications
We have added some new icons to the system and have some updates and renames. The mapping spreadsheet has been updated with this ticket. Icons mentioned in notes below are in the attached zip folder.

**[New Additions]**
- **Search similar** (16)
- **Pencil** (20)
- **Sparkline Up** (24, 20, 16)
- **Sparkline Down** (24, 20, 16)
- **Location** (16)
- **Key** (24, 16)
- **Money Back Guarantee** (24, 16) _-this is part of the "new" badge system_
- **Package** (64)
- **Thumb Up** (64)
- **Confirmation** (64)
- **Notification** (64)
- **eBay for Charity** (24, 16)
- **Transaction** (24)
- **eBay Preloved** (24, 16)
- **Selling** (16)
- **Undo** (16)
- **Jet Ski** (24, 16)
- **Snowmobile** (24, 16)

**[Updates]**
- **Search similar** (16)
- **Pencil** (20)
- **eBay Refurbished** (24, 16) _-Is now the AG checkmark_
- **Direct from Brand** (24, 16) _-Is now the AG checkmark_
- **Brand Authorized Seller** (24, 16) _-Is now the AG checkmark_
Updated 12px icons to use 12px artboards:
- **arrow left 12**
- **arrow right 12**
- **chevron down 12**
- **chevron left 12**
- **chevron right 12**
- **chevron up 12**
- **sort 12**
- **sort down 12**
- **sort up 12**

**[Renames]**
Removed **currency-** from the beginning of:
- **Dollar** (24, 16)
- **Euro** (24, 16)
- **Franc** (24, 16)
- **Krona** (24, 16)
- **Peso** (24, 16)
- **Pound** (24, 16)
- **Ringgit** (24, 16)
- **Rupee** (24, 16)
- **Zloty** (24, 16)
- **Won** (24, 16)
- **Yuan** (24, 16)

- Renamed “authenticity-guarantee-fit-” to “authenticity-guarantee-” (24, 16)
- Renamed “brand-authorized-seller-fit-” (24, 16) to “brand-authorized-seller-” (24, 16)
    - Updated artwork to the AG checkmark
- Renamed “direct-from-brand-fit-” (24, 16) to “direct-from-brand-” (24, 16)
    - Updated artwork to the AG checkmark
- Renamed “certified-recycled-fit-” (24, 16) to “certified-recycled-” (24, 16)
- Renamed “ebay-bucks-colored-16” to “ebay-bucks-logo-colored-16”
- Renamed “ebay-international-shipping-fit-” (24, 16) to “ebay-international-shipping-” (24, 16)
- Renamed “ebay-refurbished-fit-” (24, 16) to “ebay-refurbished-” (24, 16)
    - Updated artwork to the AG checkmark
- Renamed “free-warranty-16-fit”  to “free-warranty-16” -removed fit just on 16
- Renamed “nectar-24-colored”  to “nectar-logo-24-colored”
- Renamed “rotate-landscape-left” to “rotate-landscape-left-24”
- Renamed “rotate-landscape-right” to “rotate-landscape-right-24”
- Renamed “rotate-portrait-left” to “rotate-portrait-left-24”
- Renamed “rotate-portrait-right” to “rotate-portrait-right-24”
- Renamed “pause-64-colored” to “pause-filled-64-colored”
- Renamed “play-64-colored” to “play-filled-64-colored”
- Renamed “top-rated-seller-16-fit” to “top-rated-seller-16”
- Renamed “trend-up-16” to “trend-up-16-fit”
- Renamed “trend-down-16” to “trend-down-16-fit”
- Renamed “click-to-call-16-fit” to “click-to-call-16”
- Renamed “priority- (16, 24, 64)” to “attention- (16, 24, 64)”
- Renamed “priority-filled- (16, 24)” to “attention-filled- (16, 24)”
- Renamed “arrow-title-20” to “arrow-right-20” and updated artwork

Removed **social-** from the beginning of:
- **Discord** (24)
- **Facebook** (24)
- **Facebook Messenger** (24)
- **Instagram** (24)
- **Link** (24)
- **Linkedin** (24)
- **Pinterest** (24)
- **Reddit** (24)
- **Twitter** (24)
- **Whatsapp** (24)

- Renamed “arrow-title-20” to “arrow-right-20” and updated the artwork

Please let me know if you have any questions,
Ryan Tinsley

[230606_Icon_Updates.zip](https://github.com/eBay/skin/files/11668554/230606_Icon_Updates.zip)

### PR DESCRIPTION
**Pull request by ArtBlue**
Fixes #2035 

- [x] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
Various icon additions, deletions, modifications

## Notes
I created a new `deprecated` section that skips docs to differentiate between icons that need to remain and will always need to be skipped from docs and those that will be removed entirely. This will help with the icons removal with the next major version. Upon deprecated icon file removals, the deprecated entries will be removed from the full list as well as deprecated section. 

## Checklist

- [x] I verify the build is in a non-broken state
- [x] I verify all changes are within scope of the linked issue

- [x] I regenerated all CSS files under dist folder
- [x] I tested the UI in all supported browsers
- [ ] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [x] I tested the UI in dark mode and RTL mode
- [ ] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

**Review by agliga — APPROVED** [review]
LGTM
The only change I would add (but we can do it post release), is to maybe have a deprecated section in the docs in the icons section, so that teams can see which ones are deprecated 

Although, we should make use of that doc Ryan has (and maybe even link it out in our icon docs)

Either way, this PR is good to go as is

**Comment by ArtBlue** [comment]
> LGTM The only change I would add (but we can do it post release), is to maybe have a deprecated section in the docs in the icons section, so that teams can see which ones are deprecated
> 
> Although, we should make use of that doc Ryan has (and maybe even link it out in our icon docs)
> 
> Either way, this PR is good to go as is

Good suggestions. I'll look into them.

File: docs/_includes/icon.html [hunk 1]
```diff
@@ -90,4 +90,21 @@ <h3 id="icon-disabled">Disabled Icons</h3>
...
+
+    <section class="section-notice section-notice--attention" role="region" aria-label="Attention" aria-roledescription="Notice">
+        <div class="section-notice__header" id="section-notice-attention">
+            <svg class="icon icon--attention-filled-16" focusable="false" height="16" width="16" aria-label="Attention">
+                {% include symbol.html name="attention-filled-16" %}
+            </svg>
+        </div>
+        <div class="section-notice__main">
+            <h3 class="section-notice__title">Do NOT Use These!</h3>
+            <p>The following icons have been deprecated. They will be deleted in the next major Skin version. To transition usage to the new icons, please refer to the icons above and/or refer to the icon transition spreadsheet.</p>
```
Comment (LuLaValva) [inline docs/_includes/icon.html:105]: Is the icon transition spreadsheet also open source? If it is, we should link to it here.

File: docs/src/less/index.less [hunk 2]
```diff
@@ -409,6 +409,7 @@ span.demo__inner {
 
 .demo-icon-boxes li {
     align-items: center;
+    background-color: var(--color-neutral-1);
```
Comment (LuLaValva) [inline docs/src/less/index.less:412]: This doesn't work for dark mode, you should use `--color-background-secondary` instead

**Review by LuLaValva — COMMENTED** [review]
Looks great, just a quick fix for dark mode and a docs suggestion

File: docs/_includes/icon.html (hunk 1 above)
Comment (ArtBlue) [reply docs/_includes/icon.html:105]: No, I didn't include that because it is internal.

File: docs/src/less/index.less (hunk 2 above)
Comment (ArtBlue) [reply docs/src/less/index.less:412]: Great point. Good catch.

File: docs/_includes/icon.html (hunk 1 above)
Comment (LuLaValva) [reply docs/_includes/icon.html:105]: If it's internal I don't think we should mention it here. Maybe we should just say "please use equivalent icons in the section above"

Comment (ArtBlue) [reply docs/_includes/icon.html:105]: I don't think there's any danger in simply mentioning a hypothetical spreadsheet that may or may not exist. :-) A link would not be prudent, however.

**Review by LuLaValva — APPROVED** [review]

**Pull request merged**